src/coreclr/tests/src/CLRTest.CrossGen.targets

```diff
@@ -124,24 +153,55 @@ if defined RunCrossGen (
 REM CrossGen2 Script
 if defined RunCrossGen2 (
     if defined LargeVersionBubble ( set OptionalArguments=!OptionalArguments! /largeversionbubble)
-    if not exist "$(MSBuildProjectName).org" (
-        call :TakeLock
-        set CrossGen2Status=0
-        if not exist "$(MSBuildProjectName).org" (
-            mkdir IL
-            copy $(MSBuildProjectName).dll IL\$(MSBuildProjectName).dll
-            ren $(MSBuildProjectName).dll $(MSBuildProjectName).org
-            set __Command=!_DebuggerFullPath! "!CORE_ROOT!\CoreRun.exe" "!CORE_ROOT!\crossgen2\crossgen2.dll" %21scriptPath%21$(MSBuildProjectName).org -o:%21scriptPath%21$(MSBuildProjectName).dll --targetarch:x64 -O --inputbubble %21ExtraCrossGen2Args%21 -r:"!CORE_ROOT!\System.*.dll" -r:"!CORE_ROOT!\Microsoft.*.dll" -r:"!CORE_ROOT!\netstandard.dll" -r:"!CORE_ROOT!\mscorlib.dll" -r:%25cd%25\*.dll
-            echo "!__Command!"
-            call !__Command!
-            set CrossGen2Status=!ERRORLEVEL!
-        )
-        call :ReleaseLock
-        IF NOT !CrossGen2Status!==0 (
-            ECHO Crossgen2 failed with exitcode - !CrossGen2Status!
-            Exit /b 1
+    call :TakeLock
+    set CrossGen2Status=0
+    if not exist "IL" (
+        mkdir IL
+        if defined CompositeBuildMode (
+          copy *.dll IL\
+        ) else (
+          copy $(MSBuildProjectName).dll IL\
         )
     )
+
+    set __ResponseFile=!scriptPath!\composite-r2r.dll.rsp
+    del /Q !__ResponseFile!
+
+    set __Command=!_DebuggerFullPath!
+    set __Command=!__Command! "!CORE_ROOT!\CoreRun.exe"
+    set __Command=!__Command! "!CORE_ROOT!\crossgen2\crossgen2.dll"
+    set __Command=!__Command! @"!__ResponseFile!"
+    set __Command=!__Command! !ExtraCrossGen2Args!
```

Comment:
Should we echo this in to the response file - it might make reproducing issues easier if it's all in a single file we can paste into VS debug pane.

Reply:
Well, sadly this is tricky for several reasons:

1) CoreRun.exe naturally cannot be part of the response file, that's the app to run.

2) I need to double-check whether the app to run (crossgen2.dll) can be specified in a response file, probably yes.

3) I'm not sure about CoreRun parser but for instance the managed command-line parser we're using in Crossgen2, R2RDump and SuperIlc has a different syntax for arguments on the command line vs. in the response file - e.g., file arguments mustn't be quoted in response files, there's just one argument per line etc. - so that emitting !ExtraCrossGen2Args! into the response file is potentially risky.

Reply:
Sorry, I meant just the `ExtraCrossgen2Args`, so when we run crossgen2 in VS we have all the args in one place and don't have to glue extras on potentially.

Reply:
Well, yes, I need to double-check whether I can just output the extra args to the response file - in case of the managed parser, you would need to replace spaces (outside of quoted sections) with line breaks in order to make the parser work properly.

Reply:
Hmm, so it seems to me that corerun doesn't suppor response files at all and the managed parser in Crossgen2 has exactly the problem I described i.o.w. the extra command-line args have to be constructed based on knowledge whether they are going to be put directly into the command line or into a response file. I think it will be useful to switch even Crossgen1 to use response files and unify the construction of extra args everywhere but I'd prefer to do that as a separate follow-up change if you don't mind.

Reply:
Totally fine with that